### ISSUE TITLE
Different decoding of "09 F0 4C BF" compared to Hopper

### ISSUE THREAD

**Issue by mstange** [iximeow/yaxpeax-arm#3]
Hopper decodes "09 F0 4C BF" as `b.w 0x12198`.
yaxpeax-arm currently decodes it as `and pc, sb, 0x4c4c4c4c`.

```rust
fn main() {
    use yaxpeax_arch::Decoder;
    let decoder = yaxpeax_arm::armv7::InstDecoder::default_thumb();
    let mut reader = yaxpeax_arch::U8Reader::new(&[0x09, 0xF0, 0x4C, 0xBF]);
    let inst = decoder.decode(&mut reader).unwrap();
    println!("{}", inst);
}
```

I found this instruction at 0x82fc in [this libsoftokn3.so](https://github.com/mstange/profiler-get-symbols/raw/master/fixtures/android32-local/libsoftokn3.so).

**Comment by iximeow** [comment]
(i've not ignored this issue - yaxpeax-arm definitely decodes the thumb instruction incorrectly - i've just used this as motivation to add differential testing for armv7t as well, and that's taken a bit of time)

**Comment by iximeow** [comment]
ok ok, @/ehntoo put together a small slate of thumb-mode fixes, including this one. still no differential testing for thumb, but i'm clearly not getting to that any time soon :grimacing:.

in the mean time, i've merged #8 which should fix this and some other thumb issues.
